url9a01f43c15_poragchowdhury_securitygame_tgz-pJ8-securitygame_GameMasterJ8/src/securitygame/Parser.java

```diff
@@ -0,0 +1,264 @@
+package securitygame;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+public class Parser
+{
+	
+	/**
+	 * Given a .graph file, a network is generated following a predetermined format
+	 * 
+	 * @param filename
+	 * @return a Network object based on the .graph file given
+	 */
+	public static Network parseGraph(String filename)
+	{
+		boolean deleteJunkFile = false;
+		try 
+		{
+			File gFile = new File(filename);
+			if(!gFile.exists()){
+				gFile.createNewFile();
+				deleteJunkFile = true;
+			}
+			CSVParser parser = CSVParser.parse(gFile, StandardCharsets.US_ASCII, CSVFormat.DEFAULT);
+			CSVParser parseRecords= CSVParser.parse(gFile, StandardCharsets.US_ASCII, CSVFormat.DEFAULT);
+			int neighborsCounter = 0;
+			int numNodes = parseRecords.getRecords().size()/2;
+            Network network = new Network(0,numNodes);
+            network.setName(filename.substring(0,filename.indexOf(".")));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 36 does not find the substring and returns -1, the call to substring() on line 36 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string filename.